src/components/Form/TourForm.tsx

```diff
@@ -37,39 +41,50 @@ export const TourForm: React.FC<TourFormProps> = ({ userId }) => {
     debriefPlan: ''
   })
 
-  const renderTextInputs = (fields: string[]): JSX.Element[] => {
-    return fields.map(field=> {
+  const [isDepartureChecked, setDepartureCheck] = useState<boolean>(false)
 
+  const renderTextInputs = (fields: string[], prompts?: string[]): JSX.Element[] => {
+    return fields.map((field, i)=> {
       return (
         <TextField
-          value={state[field as keyof TourFormState]}
-          updateForm={e => setState({ ...state, [field]: e.target.value})}
+          key={i}
+          prompt={prompts? prompts[i] : null}
+          value={textFields[field as keyof TourFormTextFields]}
+          updateForm={e => setTextFields({ ...textFields, [field]: e.target.value})}
         />
       )
     })
   }
 
   const isChecked = (fields: string[]) => {
-    return !fields.find(field => state[field as keyof TourFormState] === '')
+    return !fields.find(field => textFields[field as keyof TourFormTextFields] === '')
```

Comment:
I think you're going to have to talk me through the logic of this one here. I know what it's responsible for and what it's doing but i'm not entirely understanding how this logic works to make the check box toggle back and forth